chapters/strings/finding-substrings.md

```diff
@@ -24,8 +24,14 @@ message.indexOf "This", 5
 message.lastIndexOf "This"
 # => 49
 
+# Count occurrences of a given string within a message
+String::occurrences = (search) ->
+  regex = new RegExp(search, 'g')
```

Comment:
To escape regexp characters try this (courtesy of [this SO answer](http://stackoverflow.com/a/3561711/227176)):

```coffee
search = search.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&")
```